### PR TITLE
Fix/plugin 1522

### PR DESCRIPTION
**Pull request by MrRahulSharma**
Converting the precision less numbers into String type and ensuring that they are compatible with the Postgres Sink as well.
Related JIRA : 
https://cdap.atlassian.net/browse/PLUGIN-1522

### CODE REVIEW

File: postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java [hunk 1]
```diff
@@ -86,14 +97,24 @@ private Object createPGobject(String type, String value, ClassLoader classLoader
   protected void writeToDB(PreparedStatement stmt, Schema.Field field, int fieldIndex) throws SQLException {
     int sqlIndex = fieldIndex + 1;
     ColumnType columnType = columnTypes.get(fieldIndex);
-    if (PostgresSchemaReader.STRING_MAPPED_POSTGRES_TYPES_NAMES.contains(columnType.getTypeName()) ||
-      PostgresSchemaReader.STRING_MAPPED_POSTGRES_TYPES.contains(columnType.getType())) {
+    if (PostgresSchemaReader.STRING_MAPPED_POSTGRES_TYPES_NAMES.contains(columnType.getTypeName())
+            || PostgresSchemaReader.STRING_MAPPED_POSTGRES_TYPES.contains(columnType.getType())) {
```
Comment (itsankit-google) [inline postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java]: nit: indentation

File: postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java [hunk 2]
```diff
@@ -57,14 +59,23 @@ protected void handleField(ResultSet resultSet, StructuredRecord.Builder recordB
...
+    int columnType = metadata.getColumnType(columnIndex);
+    if (columnType == Types.NUMERIC) {
+      int precision = metadata.getPrecision(columnIndex);
+      if (precision == 0) {
         return true;
-      default:
-        return PostgresSchemaReader.STRING_MAPPED_POSTGRES_TYPES.contains(metadata.getColumnType(columnIndex));
+      }
     }
+
```
Comment (itsankit-google) [inline postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java]: nit: remove extra empty line

File: postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresSchemaReader.java [hunk 3]
```diff
@@ -58,6 +58,14 @@ public Schema getSchema(ResultSetMetaData metadata, int index) throws SQLExcepti
       return Schema.of(Schema.Type.STRING);
     }
 
+    // If it is a numeric type without precision then use the Schema of String to avoid any precision loss
+    if (Types.NUMERIC == columnType) {
+      int precision = metadata.getPrecision(index);
+      if (precision == 0) {
+        return Schema.of(Schema.Type.STRING);
+      }
+    }
```
Comment (itsankit-google) [inline postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresSchemaReader.java]: nit: can we add a log here? similarly as we did for oracle - https://github.com/data-integrations/database-plugins/pull/323/files#diff-dae6a97a5a0b13be65e282c1b7c33fd4e995d3f1a4987ce68e5dbf5d2f1fbc9dR108

File: database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java [hunk 4]
```diff
@@ -306,7 +306,7 @@ protected void writeToDB(PreparedStatement stmt, @Nullable Schema.Field field, i
           break;
         case TIMESTAMP_MILLIS:
         case TIMESTAMP_MICROS:
-          stmt.setTimestamp(sqlIndex, Timestamp.from(record.getTimestamp(fieldName).toInstant()));
+          stmt.setTimestamp(sqlIndex, Timestamp.valueOf(record.getTimestamp(fieldName).toLocalDateTime()));
```
Comment (itsankit-google) [inline database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java]: why is this change needed?

**Review by itsankit-google — COMMENTED** [review]
can we please add some unit tests?

File: database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java (hunk 4 above)
Comment (MrRahulSharma) [reply database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java]: During the read operation in the database source plugin we convert from local to UTC, which is not getting converted back to local time during write. This issue causes the offset on DB-DB pipelines. Related JIRA : https://cdap.atlassian.net/browse/PLUGIN-1518, https://cdap.atlassian.net/browse/PLUGIN-1526

File: postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java (hunk 1 above)
Comment (MrRahulSharma) [reply postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java]: done

File: postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresSchemaReader.java (hunk 3 above)
Comment (MrRahulSharma) [reply postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresSchemaReader.java]: done

File: postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java (hunk 2 above)
Comment (MrRahulSharma) [reply postgresql-plugin/src/main/java/io/cdap/plugin/postgres/PostgresDBRecord.java]: done

**Comment by MrRahulSharma** [comment]
> can we please add some unit tests?

Added Unit Test for validating the conversion to string type.

File: database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java (hunk 4 above)
Comment (itsankit-google) [inline database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java]: Is it possible to add a unit test for this change as well please??

**Review by itsankit-google — APPROVED** [review]
LGTM

File: database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java (hunk 4 above)
Comment (MrRahulSharma) [reply database-commons/src/main/java/io/cdap/plugin/db/DBRecord.java]: Added
